### DISCUSSION TITLE
Mia-Platform console - Design section - Last used branch logic

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#64] [Q&A] [answered]
Hi guys,
I already told you that I don't like the "last used branch" feature when accessing the "Design" section, but it's another story.

Despite I don't like it, I am using it and trusting it, but I noticed that it could be not so reliable as feature.
Often I select a specific branch, I come back after a while (minutes), without logout or clean cache, but the selected branch changed to the `master` one, so I change the wrong branch.

Can you explain me the logic used by this feature?

**Comment by albertotessarotto** [comment, marked as answer]
This is the first time I have heard of this unusual behaviour happening. Surely it is a dangerous bug for those who are configuring in Console, they may not realise that the branch has changed. Let's investigate soon to replicate it. Regarding last used branch feature, it may be useful to prepare a very short survey for our users to see if this is a spread and shared opinion. Thank you for your feedback.

**Reply by frengooo** [reply]
Maybe I discovered the cause of this behavior.
This is the flow that I followed before:
- open the `Design` section
- select the `master` branch
- edit some project configuration
- save modifications, but in another branch, a new one
- the console, after the save, land the user into the `Design` section of the new branch
- open the `Deploy` section
- open the `Design` section again &rarr; the **master** branch was the selected one

**Reply by davidebianchi** [reply]
So, this seems to be a bug. Thanks for the repro steps! @albertotessarotto we should check it and fix the behaviour

**Reply by frengooo** [reply]
I confirm the repro steps, I've just tried it again 😄

**Reply by fredmaggiowski** [reply]
yep, it seems like selected branch is not updated after saving on a new branch, thanks for the info
